README.md

```diff
@@ -306,6 +307,26 @@ Given a value `m`, one can access its type representative via the
 
 1. `empty` must return a value of the same Monoid
 
+### Group
+
+A value that implements the Group specification must also implement
+the [Monoid](#monoid) specification.
```

Comment:
[There's more than one way to get to group](https://en.wikipedia.org/wiki/Magma_(algebra)#/media/File:Magma_to_group2.svg), but we definitely shouldn't let that get in the way of this PR

(We could eventually have

```
class Magma m where
  concat :: m -> m -> m

class Magma m => Quasigroup m where
  latin :: (m, m) -> (m, m)

class Magma m => Unital m where
  empty :: () -> m

class Magma m => Semigroup m

class (Semigroup m, Unital m) => Monoid m
class (Quasigroup m, Unital m) => Loop m
class (Semigroup, Quasigroup, Unital m) => Group m
```

All (non-empty) associative quasigroups are groups, so there's no 8th member)

Reply:
Oh snap. I see some possible opportunities here for another couple PRs!
Good call!